retry/context.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2016 Michal Witkowski. All Rights Reserved.
+// See LICENSE for licensing terms.
+
+package http_retry
+
+import (
+	"net/http"
+
+	"golang.org/x/net/context"
+)
+
+type ctxMarker struct{}
+
+var (
+	ctxEnableRetry = &ctxMarker{}
+)
+
+// Enable turns on the retry logic for a given request, regardless of what the retry decider says.
+//
+// Please make sure you do not pass around this request's context around.
+func Enable(req *http.Request) *http.Request {
+	if isEnabled(req.Context()) {
+		return req
+	}
+	return req.WithContext(req.Context())
```

Comment:
This should probably be `return req.WithContext(EnableContext(req.Context()))`, otherwise this function does not seem to have any effect.